pkg/crc/api/handlers.go

```diff
@@ -30,15 +30,28 @@ func stopHandler(client machine.Client, crcConfig crcConfig.Storage, _ json.RawM
 	return encodeStructToJSON(commandResult)
 }
 
-func startHandler(client machine.Client, crcConfig crcConfig.Storage, _ json.RawMessage) string {
+func startHandler(client machine.Client, crcConfig crcConfig.Storage, args json.RawMessage) string {
 	startConfig := machine.StartConfig{
-		Name:          constants.DefaultName,
-		BundlePath:    crcConfig.Get(config.Bundle).AsString(),
-		Memory:        crcConfig.Get(config.Memory).AsInt(),
-		CPUs:          crcConfig.Get(config.CPUs).AsInt(),
-		NameServer:    crcConfig.Get(config.NameServer).AsString(),
-		GetPullSecret: getPullSecretFileContent(crcConfig),
-		Debug:         true,
+		Name:       constants.DefaultName,
+		BundlePath: crcConfig.Get(config.Bundle).AsString(),
+		Memory:     crcConfig.Get(config.Memory).AsInt(),
+		CPUs:       crcConfig.Get(config.CPUs).AsInt(),
+		NameServer: crcConfig.Get(config.NameServer).AsString(),
+		Debug:      true,
+	}
+	if args != nil {
+		a := startArgs{}
+		_ = json.Unmarshal(args, &a)
+		if a == (startArgs{}) {
+			startErr := &machine.StartResult{
+				Name:  constants.DefaultName,
+				Error: "Incorrect argument name. Should be: pullSecretFile",
```

Comment:
This is conflating 2 distinct things imo
1. `pullSecretFile` is not set, while it's mandatory
2. an invalid argument was passed

When you add more arguments, this check for `invalid argument name` will no longer work,, but when you change it, we might forget to force the presence of `pullSecretFile`. If you want mark  `pullSecretFile` as mandatory, use `if a.PullSecretFile == ""`, if you want to check for invalid args, the code might need to be more involved and make use of https://golang.org/pkg/encoding/json/#Decoder.DisallowUnknownFields.
Apart from this, looks good to me.

Reply:
Thanks, modified the code, PTALA!